helix-core/src/test/java/org/apache/helix/controller/stages/MsgRecordingPerReplicaThrottleStage.java

```diff
@@ -0,0 +1,58 @@
+package org.apache.helix.controller.stages;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.helix.api.config.StateTransitionThrottleConfig;
+import org.apache.helix.controller.dataproviders.ResourceControllerDataProvider;
+import org.apache.helix.model.IdealState;
+import org.apache.helix.model.Message;
+import org.apache.helix.model.Partition;
+import org.apache.helix.model.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MsgRecordingPerReplicaThrottleStage extends PerReplicaThrottleStage {
```

Comment:
Any comment about what is this class for? Why we need a child class of PerReplicaThrottleStage instead of just adding the logic there?

Reply:
See JK's comment https://github.com/apache/helix/pull/1620#discussion_r564007432. This is to address his concern over there.

Reply:
JK's point in sum is this: Basically the test need to know the throttled msgs and their type. This logic is not needed in PerReplicaThrottleStage.java

Reply:
Anyway, for this diff, I already removed this file. We will add them back later.

Reply:
I thought we need these 2 lists for the metric reporting, no? Anyway, we can defer the discussion.